src/ima.js

```diff
@@ -1029,9 +1029,10 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
    * @memberof Ima
    */
   _maybeSaveVideoCurrentTime(): void {
-    if ((this.playOnMainVideoTag() || this.config.forceReloadMediaAfterAds) && this.player.currentTime && this.player.currentTime > 0) {
-      this.logger.debug('Custom playback used: save current time before ads', this.player.currentTime);
-      this._videoLastCurrentTime = this.player.currentTime;
+    const currentTime = this.player.getVideoElement().currentTime;
```

Comment:
```suggestion
    const currentTime = this._enigne.currentTime;
```
Let's save the real engine in `getEngineDecorator` and use it here (like we do in [imadai](https://github.com/kaltura/playkit-js-ima-dai/blob/master/src/ima-dai.js#L98))

Reply:
expose the engine on the ima.js level ?